osquery/tables/system/darwin/processes.cpp

```diff
@@ -266,6 +268,66 @@ void genProcNumThreads(QueryContext& context, int pid, Row& r) {
   }
 }
 
+uint64_t parseKCItem(kcdata_item_t item) {
+  if (item->type == STACKSHOT_KCTYPE_TASK_SNAPSHOT) {
+    struct task_snapshot_v2* tsv2 = (struct task_snapshot_v2*)item->data;
+    return tsv2->ts_unique_pid;
+  } else {
+    return ULLONG_MAX;
+  }
+}
+
+void genProcUniquePid(QueryContext& context, int pid, Row& r) {
+  if (!context.isColumnUsed("unique_pid")) {
+    return;
+  }
+
+  // A definition of the stackshot_config structure so we can pull data out of
+  // the kernel
+  typedef struct stackshot_config {
+    int pid;
+    uint32_t flags;
+    uint64_t dtimestamp;
+    uint64_t buf;
+    uint32_t buf_size;
+    uint64_t out_buf_addr;
+    uint64_t out_buf_size_addr;
+  } stackshot_config_t;
+
+  char* addr = nullptr;
+  uint64_t size = 0;
+
+  stackshot_config_t stconf = {0};
+  stconf.pid = pid;
+  stconf.buf = 0;
+  stconf.buf_size = 0;
+  stconf.flags = 0x10000; // STACKSHOT_KCDATA_FORMAT
+  stconf.out_buf_addr = (uint64_t)&addr;
```

Comment:
C-cast is a tricky thing. Any cast could be under it. Also it is hard to notice and grep in the code.
You could use a `reinterpret_cast` here instead.